src/System.IO.Pipelines/src/System/IO/Pipelines/SynchronizationContextPipeScheduler.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Diagnostics;
+using System.Threading;
+
+namespace System.IO.Pipelines
+{
+    internal class SynchronizationContextPipeScheduler : PipeScheduler
+    {
+        public override void Schedule<TState>(Action<TState> action, TState state)
+        {
+            Debug.Assert(SynchronizationContext.Current != null);
+
+            SynchronizationContext.Current.Post(state => action(state), null);
```

Comment:
This will allocate both a delegate and a closure.  You could limit that to just a single manual closure with code like:
```C#
_ctx.Post(s =>
{
    var t = (Tuple<Action<TState>,TState>)s;
    t.Item1(t.Item2);
}, Tuple.Create(action, state));
```

Reply:
Or if you're going to allocate a SynchronizationContextPipeScheduler per operation that needs one anyway, you can just store the action and state as fields onto that instance, and then pass in `this`, and reuse the allocation rather than needing more.

Reply:
Good idea. Will do that.